src/problems/problemData.ts

```diff
@@ -34,16 +36,34 @@ export const courseIdToProgramIdLists: Record<CourseId, ProgramId[][]> = {
   tuBeginner2: [],
 };
 
-export function generateProgram(programId: ProgramId, languageId: LanguageId): string {
+export function generateProgram(programId: ProgramId, languageId: LanguageId): GeneratedProgram {
   // TODO(exKAZUu): 問題IDに紐づくプログラム（テンプレート）を取得して、乱数を使って具体的なプログラムを生成する。
-  return (
-    `const character1 = new Character();
-character1.moveForward();
-character1.moveForward();
-character1.moveForward();
-character1.moveForward();
-character1.moveForward();` || programIdToLanguageIdToProgram[programId][languageId]
+  console.log('languageId', languageId);
+  const randomNumberRegex = /<(\d+)-(\d+)>/g;
+  const programTemplete = programIdToLanguageIdToProgram[programId];
+  const jsTemplete = programTemplete['js'];
+  const randomNumberArray: number[] = [];
+  const jsProgram = jsTemplete.replaceAll(randomNumberRegex, (match, min, max) => {
+    const randomNumber = getRandomInt(Number(min), Number(max));
+    randomNumberArray.push(randomNumber);
+    return randomNumber.toString();
+  });
+
+  let index = 0;
+  const displayProgram = programTemplete['java'].replaceAll(randomNumberRegex, () =>
+    randomNumberArray[index++].toString()
   );
```

Comment:
> 現状乱数は生成できるのですが、langageIdをうまく取ってくることができません。

以下のようにするのはどうですか？（languageId がなかったときはエラーにならないように空を返す）

```suggestion
  const displayProgram = languageId ? programTemplete[languageId].replaceAll(randomNumberRegex, () =>
    randomNumberArray[index++].toString()
  ) : '';
```
languageId が空で来てしまうのは、画面のレンダリングが複数行われて、langageIdがsession storageから取得できていない時に generateProgram が呼ばれているからだと思います。3回目のレンダリングで取得はできるので、上記のようにして取得できていないときはエラーにならないようにすれば良いのではと思いました。